internal/providers/vault/issue_257.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 Namespace Labs Inc; All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+
+package vault
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/vault-client-go"
+)
+
+// Temporary workaround until the fix is merged:
+// https://github.com/hashicorp/vault-client-go/pull/260
+func withIssue257Workaround() vault.ClientOption {
+	c := vault.DefaultConfiguration().HTTPClient
+	c.Transport = fix257{rt: c.Transport}
+	return vault.WithHTTPClient(c)
+}
+
+type fix257 struct {
+	rt http.RoundTripper
+}
+
+func (w fix257) RoundTrip(req *http.Request) (*http.Response, error) {
+	res, err := w.rt.RoundTrip(req)
+	if !strings.HasSuffix(req.URL.Path, "/secret-id") ||
+		err != nil ||
+		res == nil ||
+		res.StatusCode != http.StatusOK ||
+		strings.Split(res.Header.Get("content-type"), ";")[0] != "application/json" {
+		return res, err
+	}
+	defer res.Body.Close()
+
+	body := map[string]any{}
+	json.NewDecoder(res.Body).Decode(&body)
```

Comment:
error checking?

If you want to drop errors, please do so explicitly, e.g.:

```
// Swallow the decoding error as we interpret it as a missing data field.
_ = json.NewDecoder(...).Decode(...)
```

Reply:
RoundTrip should not error based on the content… But yes, we should swallow the error, but this isn't good as-is, because in this case we would consume the content.

Let me change this.